### ISSUE TITLE
Grunt Misusing Logging System

### ISSUE THREAD

**Issue by robertmain** [balderdashy/sails#1563]
Grunt seems to be misusing the logging system to show deprecation warnings for SASS compilation(ie: calling sails.log.error rather than sails.log.warn). I would fix it myself and send a pull request, but I can't find out where it's taking place.
Apologies if this is the wrong place to file this issue.

**Comment by mikermcneil** [comment]
@tjwebb @sgress454 @irlnathan @particlebanana @rachaelshaw @loicsaintroch @jpwilliams @gtg092x @robertmain hey guys, here's what's up as far as I see it:

The Grunt hook currently sniffs out how serious a particular log looks by sniffing stderr and stdout from the childproc using a regexp.  It does this because this is the best way Grunt has (at least the version we're using) to access output programmatically (i.e. to avoid spewing logs all over the place)  Undoubtedly this is where everyone chimes in about gulp :p

I need your help:

1) What do you want to see in the future out of grunt/gulp support? (i.e. please don't tell me which one you like better-- rather, what do you need for the stuff you're working on?)
2) What grunt/gulp modules do you use most often?  LESS?  SASS?  Linker? live reload? etc.

Thank you!

**Comment by robertmain** [comment]
I would really like gulp support in the future and I actually [currently have a gulp setup](https://bitbucket.org/elementalsoftware/pillars) that I was planning to port to sails at some point or submit a pull request for. I'm currently cutting my teeth on SASS and native support for this in Sails would be great. In particular - support for the ability to swap SASS "frameworks" (like bourbon, compass etc) in and out using bower(you'll notice this problem is already solved in the bitbucket repo I linked earlier) as I don't actually care for Compass and prefer Bourbon instead

**Comment by masumeEbhami** [comment]
I almost use SASS modules .but i dont know about future. becaues every time is maybe that one person break rules and introduce a amazing thing that is better than grunt/gulp.:-)

**Comment by loicsaintroch** [comment]
Sails tried for a long time to take care of the front-end workflow with some default Grunt tasks, pipeline and template engines. This is great if you focus on it but this is not the case here. In my opinion, Sails shouldn't take care of the front-end at all. Every front-end developer has their own preferences and they will never be happy with a default workflow.

**Comment by robertmain** [comment]
... Actually, I think I'm inclined to agree with @loicsaintroch. Maybe sails could ship(pun not intended) with NO frontend asset pipeline at all and then have optional node packages that you can install to set sails up for gulp/grunt/brocolli/whatever

**Comment by mikermcneil** [comment]
@robertmain @loicsaintroch @masumeEbhami thanks for the feedback guys.  I've heard the sentiment before and tend to agree-- it would be best if we got out of your way altogether and shipped a lighter version of Sails.  Lots of advantages here outside of just Grunt, imo, that we ought to discuss somewhere else.

Anyways, all that said, many folks come to the framework expecting to find some kind of solution for the asset pipeline, so we want to make it easy to install.  I have some ideas here but want to get more feedback from everyone first, if you'd be so kind :) 

Quick aside-- just want to point out that @rwhitmire did [some great work on linker](https://twitter.com/badthingfactory/status/587378048279916545) to make it more flexible.  i.e. goal is to remove the need for pipeline.js.

**Comment by CWyrtzen** [comment]
Any more feedback here?

**Comment by robertmain** [comment]
It would be great to have grunt removed by default and just have a generator command for putting it back(for folks that want the default workflow)

**Comment by Karnith** [comment]
I agree, then gulp could be implemented for those that want it. I've created a machine pack in the meantime that could possible be used as an example of this with a toggle to switch between gulp and grunt. https://github.com/Karnith/machinepack-sailsgulpify

**Comment by sailsbot** [comment]
Thanks for posting, @robertmain.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
